File: python/updating.py
#!/usr/bin/env python
from __future__ import print_function

from couchbase.bucket import Bucket
import couchbase.exceptions as E

cb = Bucket('couchbase://10.0.0.31/default')

# This always works!
print('Upserting')
cb.upsert('docid', {'property': 'value'})
print('Getting item back. Value is:',
      cb.get('docid').value)
print('...')

print('Will try to insert the document. Should fail because the item already exists..')
try:
    cb.insert('docid', {'property': 'value'})
except E.KeyExistsError:
    print('Insert failed because item already exists!')
print('...')

print('Replacing the document. This should work because the item already exists')
cb.replace('docid', {'property': 'new_value'})
print('Getting document again. Should contain the new contents:',
      cb.get('docid').value)
print('...')

print('Removing document.')
# Remove the item, then try to replace it!
cb.remove('docid')
print('Replacing document again. Should fail because document no longer exists')
try:
    cb.replace('docid', {'property': 'another value'})
except E.NotFoundError:
    print('Get failed since item does not exist')

File: php/updating.php
<?php

$cluster = new \Couchbase\Cluster("couchbase://localhost");
$bucket = $cluster->openBucket("default");

// This always works!
echo "Upserting\n";
$bucket->upsert('docid', array('property' => 'value'));
echo "Getting item back...";
echo "Value is: \n";
print_r($bucket->get('docid')->value);
echo "\n....\n";

echo "Inserting..\n";
echo "Will try to insert the document...\n";
echo "Should fail because the item already exists..\n";
try {
    $bucket->insert('docid', array('property' => 'value'));
    throw new Exception("Shouldn't reach here!");
} catch (\Couchbase\Exception $ex) {
    printf("Got error: Code=0x%x, Message=%s\n", $ex->getCode(), $ex->getMessage());
}
echo "...\n";

echo "Replacing the document. This should work because the item already exists";
$bucket->replace('docid', array("property" => "new_value"));
echo "Getting document again. Should contain the new contents:\n";
print_r($bucket->get('docid')->value);
echo "\n...\n";

echo "Removing document";
$bucket->remove('docid');
echo "Replacing document again. Should fail because document no longer exists\n";

try {
    $bucket->replace("docid", array("property" => "new_value"));
} catch (\Couchbase\Exception $ex) {
    printf("Got error %s\n", $ex->getMessage());
}